### PR TITLE
Implementation of undelete function in web api

### PR DESCRIPTION
**Pull request by juleengraham**
impletementing undelete function in webapps api to allow previously deleted users, who had not been purged, to be undeleted.

### CODE REVIEW

**Comment by martenson** [comment]
@galaxybot test this

File: lib/galaxy/webapps/galaxy/api/users.py [hunk 1]
```diff
@@ -242,17 +242,36 @@ def delete(self, trans, id, **kwd):
...
+            self.user_manager.delete(user)
         return self.user_serializer.serialize_to_view(user, view='detailed')
 
     @expose_api
     @web.require_admin
-    def undelete(self, trans, **kwd):
-        raise exceptions.NotImplemented()
+    def undelete(self, trans, id, **kwd):
+        """
+        UNDELETE /api/users/{id}
```
Comment (jmchilton) [inline lib/galaxy/webapps/galaxy/api/users.py]: DELETE here is an HTTP method/verb (https://www.restapitutorial.com/lessons/httpmethods.html), UNDELETE isn't an option. Did you use a route with POST/PUT/etc.. to test this or is it untested?

**Comment by martenson** [comment]
@galaxybot test this

**Comment by martenson** [comment]
The result of CI lint test is relevant here. (you can get to it by expanding the checks and clicking on 'Detail' of the failing one.

```
./test/api/test_users.py:79:9: F841 local variable 'response' is assigned to but never used
```

**Comment by martenson** [comment]
continues at https://github.com/galaxyproject/galaxy/pull/8299

**Pull request closed without merging**